### PR TITLE
ADT: Bug fix: serialization issue with strings which starts with a numeric value. 

### PR DESCRIPTION
**Pull request by azabbasi**
We have a custom serialization layer which makes sure that a string is a Json payload and based on that logic writes the value into the json generator.
The logic where we decide whether or not a string value is a valid json, automatically decides that a string value that starts with a number is a valid json payload and writes the value which is a number to the json generator. 

By making sure that either the input string is a container node or is a simple text, we can avoid this issue.

### CODE REVIEW

**Review by timtay-microsoft — APPROVED** [review]

File: sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/implementation/serializer/DigitalTwinsStringSerializer.java
Comment (abhipsaMisra) [inline sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/implementation/serializer/DigitalTwinsStringSerializer.java:38]: q - is this to ensure primitive values are not treated as json (numeric, bool etc)? Can jsonInString be an array?

Comment (azabbasi) [reply sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/implementation/serializer/DigitalTwinsStringSerializer.java:38]: That is correct, Arrays are fine, since only strings will end up here, boolean tokens will also fall into the same category, if someone decides to use "true" or "false" as string values, the previous code would treat them as actual boolean tokens as they are valid json tokens. by making sure that only text or container nodes go through, we will maintain the functionality where we accept the full json payload and do not write json values when they are non-string primitive types.

File: sdk/digitaltwins/azure-digitaltwins-core/CHANGELOG.md
Comment (drwill-ms) [inline sdk/digitaltwins/azure-digitaltwins-core/CHANGELOG.md:3]: Did we ship a 1.1.0 on 2021-05-13?

File: sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/implementation/serializer/DigitalTwinsStringSerializer.java
Comment (drwill-ms) [inline sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/implementation/serializer/DigitalTwinsStringSerializer.java:37]: Could use some code comments for future maintainers about what is valid, what is invalid, and why

File: sdk/digitaltwins/azure-digitaltwins-core/src/test/java/com/azure/digitaltwins/core/TwinAsyncTests.java
Comment (drwill-ms) [inline sdk/digitaltwins/azure-digitaltwins-core/src/test/java/com/azure/digitaltwins/core/TwinAsyncTests.java:481]: It looks like we have 2 positive test cases added, but no negative test cases.

File: sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/implementation/serializer/DigitalTwinsStringSerializer.java
Comment (drwill-ms) [inline sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/implementation/serializer/DigitalTwinsStringSerializer.java:35]: Is there any way to unit test this? It feels to me it would be more efficient to be able to do that, rather than have to construct models, etc.

File: sdk/digitaltwins/azure-digitaltwins-core/CHANGELOG.md
Comment (timtay-microsoft) [reply sdk/digitaltwins/azure-digitaltwins-core/CHANGELOG.md:3]: No. We had planned to, but it got delayed since we needed approval from the Azure SDK team. I'll update this changelog to have today's date before I go through with the release

File: sdk/digitaltwins/azure-digitaltwins-core/src/test/java/com/azure/digitaltwins/core/TwinAsyncTests.java
Comment (azabbasi) [reply sdk/digitaltwins/azure-digitaltwins-core/src/test/java/com/azure/digitaltwins/core/TwinAsyncTests.java:481]: I will write unit tests for the serializer and capture the negative test cases. how does that sound?

File: sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/implementation/serializer/DigitalTwinsStringSerializer.java
Comment (azabbasi) [reply sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/implementation/serializer/DigitalTwinsStringSerializer.java:35]: I like that idea, I will add unit tests for this.

**Review by drwill-ms — APPROVED** [review]

**Pull request merged**
